lib/Service/Mail/Mailer.hs

```diff
@@ -4,64 +4,148 @@ module Service.Mail.Mailer
   , sendResetPasswordMail
   ) where
 
-import Control.Exception (SomeException, catch)
+import Control.Exception (SomeException, catch, handle)
 import Control.Lens ((^.))
 import Control.Monad.Reader (asks, liftIO)
+import qualified Data.Aeson as Aeson
 import qualified Data.ByteString as S
 import qualified Data.ByteString.Lazy as B
+import Data.Either (rights)
+import Data.HashMap.Strict (HashMap, fromList)
 import qualified Data.Text as T
+import qualified Data.Text.Encoding as E
 import qualified Data.Text.Lazy as TL
 import qualified Data.UUID as U
 import qualified Network.HaskellNet.Auth as Auth
 import qualified Network.HaskellNet.SMTP as SMTP
 import qualified Network.HaskellNet.SMTP.SSL as SMTPSSL
 import qualified Network.Mail.Mime as MIME
 import qualified Network.Mail.SMTP as SMTPMail
+import qualified Network.Mime as MIME
+import System.Directory (listDirectory)
+import System.FilePath (takeFileName)
 
 import Constant.Component
 import LensesConfig
 import Localization
 import Model.Context.AppContext
 import Model.User.User
 import Util.Logger
+import Util.Template (loadAndRender)
 
-sendRegistrationConfirmationMail :: Email -> U.UUID -> String -> AppContextM ()
-sendRegistrationConfirmationMail email userId hash = do
+sendRegistrationConfirmationMail :: User -> U.UUID -> String -> AppContextM ()
+sendRegistrationConfirmationMail user userId hash = do
```

Comment:
Can we take `userId` from `user` object?

Reply:
OK, good point... If it is the same, I can improve that - I was a bit confused here:

https://github.com/ds-wizard/dsw-server/blob/754321faf0eb1015915b5ec00420c6f122ea467c/lib/Service/User/UserService.hs#L67

It uses `(actionKey ^. userId)` and not `uUuid` nor `(user ^. uuid)`...

Reply:
Or maybe it would give more sense to build the link(s) in `UserService` instead of `Mailer` and pass it just as a `String`. The knowledge of link lies in `UserService` right?

Reply:
https://github.com/ds-wizard/dsw-server/pull/45/commits/38a98bf1eb7538a2b961fb9cca15b0760ae8fb7c